src/analysis/fourier/add_circle.lean

```diff
@@ -266,64 +286,114 @@ end
 
 end monomials
 
-section fourier
+section scope_hT -- everything from here on needs `0 < T`
 variables [hT : fact (0 < T)]
 include hT
 
-/-- We define `fourier_series` to be a `ℤ`-indexed Hilbert basis for `Lp ℂ 2 haar_add_circle`,
+
+section fourier_coeff
+
+/-- The Fourier coefficients of a function `add_circle T → ℂ`, defined as the integral over
+`add_circle T` of `λ t, fourier (-n) t * f t`. -/
+def fourier_coeff (f : add_circle T → ℂ) (n : ℤ) : ℂ :=
```

Comment:
can you define these for a function taking values in a general complete complex vector space?

Reply:
Sure. I'm afraid that we won't be able to prove anything nontrivial for any coefficients other than `ℂ` without a considerable amount more work, though.